crates/state/src/tx.rs

```diff
@@ -41,18 +43,46 @@ pub struct DepositRequestInfo {
     pub address: Vec<u8>,
 }
 
+/// Wrapper to hold various types of Inscription data as defined by [`BlobType`] enum.
 #[derive(Clone, Debug, PartialEq, Eq, BorshSerialize, BorshDeserialize, Arbitrary)]
-pub struct InscriptionData {
+pub struct InscriptionBlob {
+    /// for tagging purpose to understand what kind of inscription it is
+    data_type: BlobType,
     /// payload present in inscription transaction (either batchTx or checkpointTx)
-    batch_data: Vec<u8>,
+    data: Vec<u8>,
 }
 
-impl InscriptionData {
-    pub fn new(batch_data: Vec<u8>) -> Self {
-        Self { batch_data }
+/// Enum that acts as a tag to separates different types of inscription blobs.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, BorshSerialize, BorshDeserialize, Arbitrary)]
+#[borsh(use_discriminant = true)]
+pub enum BlobType {
+    Checkpoint = 0,
+    DA = 1,
+}
+
+impl BlobType {
+    /// Used to convert Integer to Type mainly for parsing purpose
+    pub fn from_u32(value: u32) -> Option<Self> {
+        match value {
+            0 => Some(Self::Checkpoint),
+            1 => Some(Self::DA),
+            _ => None,
+        }
+    }
+}
```

Comment:
Needs a method like:

```rs
pub fn get_tag_bytes(&self, params: &RollupParams) -> &[u8] {
    match self {
        Self::Checkpoint => params.envelope_ckpt_tag.as_bytes()
        Self::DA => params.envelope_da_tag.as_bytes()
    }
}
```

Of course, those fields might be encapsulated in a different structure, but that's the idea.